spec/lib/metasploit/framework/credential_collection_spec.rb

```diff
@@ -311,6 +313,20 @@
           Metasploit::Framework::Credential.new(public: username, private: password),
         )
       end
+
+      context "when using password spraying" do
+        let(:password_spray) { true }
+
+        # REMOVE BEFORE COMMIT: yields nothings, fails because of bug in method
+        context "without password" do
```

Comment:
What would your thoughts be on not nesting the `context`s here. My thinking is it would make searching for failing tests much harder, as the sting would be broken across multiple lines whereas if it was a single context it would be much easier to find.

Here is where I usually check "best practices" when writing tests, just in case it's useful:
https://www.betterspecs.org/#contexts

Reply:
Ok with me 👍 
I struggled a bit to find a proper organization of the cases, given the already existing specs.

Do you want to merge only those two, or do you think I should also do the same for the other cases?
(meaning having only 1-level deep contexts)

Reply:
> Do you want to merge only those two, or do you think I should also do the same for the other cases? (meaning having only 1-level deep contexts)

Yea, I personally think this would help make things clearer.

Reply:
Ok, fine with me. I'll change that!

Reply:
This was a bit painful to do but I agree, once it is done, it is much clearer to read and to debug 🎉